test/csuite/schema_abort/main.c

```diff
@@ -692,6 +692,11 @@ thread_run(void *arg)
             stable_ts = __wt_atomic_addv64(&global_ts, 1);
         testutil_check(__wt_snprintf(kname, sizeof(kname), "%" PRIu64, i));
 
+        if (0) {
+retry:
```

Comment:
Another option is to restructure this as a loop so you can avoid jumping upwards like this + get rid of the macro.

Reply:
I'll have to fix test_timestamp_abort too so I'll try to arrive at some kind of general case.

Reply:
Agreed - the goto is a bit gross.

Reply:
While the goto might seem gross, using a for loop seems worse to me. You can't wrap the calls in macros and have to handle the potential output individual if statements, you also end up wrapping a lot of stuff in the for loop, and then you break out of the for loop at the end which is more confusing than a goto retry.

Heres a patch of the for loop version with the indentation kept the same to reduce the number of changes:
```
+++ b/test/csuite/schema_abort/main.c
@@ -582,11 +582,12 @@ thread_run(void *arg)
 {
     FILE *fp;
     WT_CURSOR *cur_coll, *cur_local, *cur_oplog;
+    WT_DECL_RET;
     WT_ITEM data;
     WT_RAND_STATE rnd;
     WT_SESSION *oplog_session, *session;
     THREAD_DATA *td;
-    uint64_t i, stable_ts;
+    uint64_t i, stable_ts, retry;
     char cbuf[MAX_VAL], lbuf[MAX_VAL], obuf[MAX_VAL];
     char kname[64], tscfg[64];
     bool use_prep;
@@ -692,11 +693,7 @@ thread_run(void *arg)
             stable_ts = __wt_atomic_addv64(&global_ts, 1);
         testutil_check(__wt_snprintf(kname, sizeof(kname), "%" PRIu64, i));

-        if (0) {
-retry:
-            testutil_check(session->rollback_transaction(session, NULL));
-        }
-
+    for (retry = 0; retry < 100; retry ++) {
         testutil_check(session->begin_transaction(session, NULL));
         if (use_prep)
             testutil_check(oplog_session->begin_transaction(oplog_session, NULL));
@@ -715,11 +712,21 @@ retry:
         data.size = __wt_random(&rnd) % MAX_VAL;
         data.data = cbuf;
         cur_coll->set_value(cur_coll, &data);
-        testutil_check_rollback_retry(cur_coll->insert(cur_coll));
+        if ((ret = cur_coll->insert(cur_coll) == WT_ROLLBACK) {
+            testutil_check(session->rollback_transaction(session, NULL));
+            continue;
+        } else {
+            testutil_die(ret);
+        }
         data.size = __wt_random(&rnd) % MAX_VAL;
         data.data = obuf;
         cur_oplog->set_value(cur_oplog, &data);
-        testutil_check_rollback_retry(cur_oplog->insert(cur_oplog));
+        if ((ret = cur_oplog->insert(cur_oplog)) == WT_ROLLBACK) {
+            testutil_check(session->rollback_transaction(session, NULL));
+            continue;
+        } else {
+            testutil_die(ret);
+        }
         if (use_ts) {
             /*
              * Run with prepare every once in a while. And also yield after prepare sometimes too.
@@ -759,6 +766,8 @@ retry:
             if (use_prep)
                 testutil_check(oplog_session->commit_transaction(oplog_session, NULL));
         }
+        break;
+    }
         /*
          * Insert into the local table outside the timestamp txn.
          */

```